File: recursion/basic_recursion/reverse_array/reverse.cpp
#include <bits/stdc++.h>
using namespace std;

void printarray(int arr[], int n){
    cout<<"The reversed array is: "<<endl;
    for(int i=0; i<n; i++){
        cout<<arr[i]<<" ";
    }
}

void reversed(int arr[], int start , int end){
    if(start<end){
        swap(arr[start], arr[end]);
        reversed(arr, start+1, end-1);
    }
}

int main(){
    int n;
    cout<<"Enter the size of the array: ";
    cin>>n;
    int arr[n];
    for(int i=0; i<n; i++){
        cin>>arr[i];
    }
    reversed(arr, 0, n-1);
    printarray(arr, n);
}

File: recursion/basic_recursion/reverse_array/reverse.py
def reversed(arr, start, end):
    if(start >= end):
        return arr
    else:
        arr[start], arr[end] = arr[end], arr[start]
        return reversed(arr, start+1, end-1)

n=int(input("Enter the number of numbers in array: "))
arr = []
i = 0
while i<n:
    arr.append(int(input("Enter the number: ")))
    i+=1

print("The reversed array is: ", reversed(arr, 0, n-1))